instrumentor/controllers/instrumentationconfig/workloads_controllers.go

```diff
@@ -0,0 +1,89 @@
+package instrumentationconfig
+
+import (
+	"context"
+
+	odigosv1alpha1 "github.com/odigos-io/odigos/api/odigos/v1alpha1"
+	"github.com/odigos-io/odigos/common/consts"
+	"github.com/odigos-io/odigos/k8sutils/pkg/utils"
+	"github.com/odigos-io/odigos/k8sutils/pkg/workload"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// These controllers handle update of the InstrumentationConfig's ServiceName
+// whenever there are changes in the associated workloads (Deployments, DaemonSets, StatefulSets).
+
+type DeploymentReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+func (r *DeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	return reconcileWorkload(ctx, r.Client, workload.WorkloadKindDeployment, req)
+}
+
+type DaemonSetReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+func (r *DaemonSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	return reconcileWorkload(ctx, r.Client, workload.WorkloadKindDaemonSet, req)
+}
+
+type StatefulSetReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+func (r *StatefulSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	return reconcileWorkload(ctx, r.Client, workload.WorkloadKindStatefulSet, req)
+}
+
+func reconcileWorkload(ctx context.Context, k8sClient client.Client, objKind workload.WorkloadKind, req ctrl.Request) (ctrl.Result, error) {
+	obj := workload.ClientObjectFromWorkloadKind(objKind)
+	instConfigName := workload.CalculateWorkloadRuntimeObjectName(req.Name, objKind)
+
+	if err := getWorkloadObject(ctx, k8sClient, req, obj); err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	annotations := obj.GetAnnotations()
+	reportedName := annotations[consts.OdigosReportedNameAnnotation]
```

Comment:
Can we replace this entire code section with `resolveServiceName`?

It does just that, right?